### PR TITLE
Update CLI Warp deployment output shape

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

- Update CLI Warp deployment output shape to new WarpCore config
- Clarify that CLI warp config example is for deployment

### Related issues

Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3298

### Backward compatibility

No, rollout should correspond with Warp UI update: https://github.com/hyperlane-xyz/hyperlane-warp-ui-template/pull/134

### Testing

Executed a basic warp route deployment

### CODE REVIEW

File: typescript/cli/examples/warp-deployment.yaml
Comment (paulbalaji) [inline typescript/cli/examples/warp-deployment.yaml:1]: when you pull latest main, you will need to update the filename of `typescript/cli/examples/fork/warp-tokens.yaml` as well

File: typescript/cli/src/config/warp.ts [hunk 1]
```diff
@@ -44,17 +43,17 @@ export const WarpRouteConfigSchema = z.object({
...
   base: InferredType['base'];
   synthetics: Array<InferredType['synthetics'][0]>;
 };
 
-export function readWarpRouteConfig(filePath: string) {
+export function readWarpDeployConfig(filePath: string) {
   const config = readYamlOrJson(filePath);
-  if (!config) throw new Error(`No warp config found at ${filePath}`);
-  const result = WarpRouteConfigSchema.safeParse(config);
+  if (!config) throw new Error(`No warp deploy config found at ${filePath}`);
```
Comment (paulbalaji) [inline typescript/cli/src/config/warp.ts:55]: in some places we use "warp deploy config" and others "warp route deployment config", think it makes sense to unify everything to the latter form for clarity

Comment (paulbalaji) [reply typescript/cli/src/config/warp.ts:55]: probably want to do that with variable/function naming too 
e.g. 
- warpDeploymentConfigPath -> warpRouteDeploymentConfigPath
- createWarpDeployConfigCommand -> createWarpRouteDeploymentConfigCommand

thoughts?

File: typescript/cli/src/config/warp.ts [hunk 2]
```diff
@@ -114,12 +113,12 @@ export async function createWarpConfig({
...
   };
 
-  if (isValidWarpRouteConfig(result)) {
+  if (isValidWarpDeployConfig(result)) {
     logGreen(`Warp Route config is valid, writing to file ${outPath}`);
     writeYamlOrJson(outPath, result, format);
   } else {
     errorRed(
-      `Warp config is invalid, please see https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/cli/examples/warp-tokens.yaml for an example`,
+      `Warp config is invalid, please see https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/cli/examples/warp-deployment.yaml for an example`,
```
Comment (paulbalaji) [inline typescript/cli/src/config/warp.ts:121]:
```suggestion
      `Warp Route deployment config is invalid, please see https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/cli/examples/warp-deployment.yaml for an example`,
```

File: typescript/cli/src/deploy/warp.ts [hunk 3]
```diff
@@ -323,43 +317,46 @@ function writeTokenDeploymentArtifacts(
...
-      hypNativeAddress: hypTokenAddr,
-    };
-  } else {
-    throw new Error(`Unsupported token type: ${baseConfig.type}`);
+
+  // Second pass, add connections between tokens
+  // Assumes full interconnectivity between all tokens for now b.c. that's
+  // what the deployers do by default.
+  for (const token1 of warpCoreConfig.tokens) {
+    for (const token2 of warpCoreConfig.tokens) {
```
Comment (paulbalaji) [inline typescript/cli/src/deploy/warp.ts:344]: interoperability ftw!

**Review by paulbalaji — APPROVED** [review]
logic lgtm, left some thoughts on naming consistency

File: typescript/cli/src/config/warp.ts (hunk 1 above)
Comment (jmrossy) [reply typescript/cli/src/config/warp.ts:55]: Agreed, I'll do that now

**Pull request merged**